ts/DataGrid/DataGrid2/DataGrid.ts

```diff
@@ -133,6 +133,15 @@ class DataGrid {
     *
     * */
 
+    /**
+     * Destroys the data grid.
+     */
+    public destroy(): void {
+        this.viewport.destroy();
+        this.tableElement.remove();
+        this.container.classList.remove(Globals.classNames.container);
```

Comment:
When you destroy the DataGrid and log what is left, you still get a few properties which should not be there:
![image](https://github.com/highcharts/highcharts/assets/57004120/aa33461d-3ab1-4fc9-8ade-a74b1f692558)